Formula/lib/libpq@16.rb

```diff
@@ -0,0 +1,79 @@
+class LibpqAT16 < Formula
+  desc "Postgres C API library"
+  homepage "https://www.postgresql.org/docs/16/libpq.html"
+  url "https://ftp.postgresql.org/pub/source/v16.8/postgresql-16.8.tar.bz2"
+  sha256 "9468083a56ce0ee7d294601b74dad3dd9fc69d87aff61f0a9fb63c813ff7efd8"
+  license "PostgreSQL"
+
+  livecheck do
+    url "https://ftp.postgresql.org/pub/source/"
+    regex(%r{href=["']?v?(\d+(?:\.\d+)+)/?["' >]}i)
+  end
+
```

Comment:
since it is gonna be deprecated on 09 Nov 2028, might worth adding a note in here

https://endoflife.date/postgresql